pkg/k8s/cnp.go

```diff
@@ -213,6 +213,10 @@ retryLoop:
 		// most likely not temporary.
 		serverRule, err = c.prepareUpdate(cnp, scopedLog)
```

Comment:
this will also return an error if the CNP does not exist in the cache.Store, we don't want to update the status for that case. I suggest you create an error on top of this file
```
type ErrParse struct{
 msg string
}

func (e ErrParse) Error() string{
 return e.msg
}

func IsErrParse(e error) bool{
    _, ok := e.(ErrParse)
   return ok
}

// then in the c.prepareUpdate(cnp, scopedLog) when you return an error
return ErrParse{msg: fmt.Sprintf("invalid CNP %s", err)}

// and here you would do
serverRule, err = c.prepareUpdate(cnp, scopedLog)
if IsErrParse(err) {
    statusErr := c.updateStatus(serverRule, rev, err, waitForEPsErr)
....
}
```

Reply:
Fixed, thanks!